src/App.tsx

```diff
@@ -14,210 +34,170 @@ export const App: React.FC = () => {
     }
   }, []);
 
+  const loadTodos = async () => {
+    if (!user) {
+      return;
+    }
+
+    const todosFromServer = await getTodos(user.id);
+
+    setTodos(todosFromServer);
+  };
+
+  useEffect(() => {
+    loadTodos();
+  }, [user]);
+
+  const removeError = useCallback(() => {
+    setError('');
+  }, []);
+
+  const handleError = useCallback((errorText: string) => {
+    setError(errorText);
+    setTimeout(removeError, 3000);
+  }, []);
+
+  const addTodo = useCallback(async (title: string) => {
+    if (title.trim().length === 0) {
+      handleError('Title can\'t be empty');
+    }
+
+    if (user && title.trim().length > 0) {
+      setIsAdding(true);
+
+      try {
+        await postTodo({
+          userId: user.id,
+          title,
+          completed: false,
+        });
+
+        await loadTodos();
+      } catch {
+        handleError('Unable to add a todo');
+      }
+
+      setIsAdding(false);
+      setTodoTitle('');
+    }
+  }, []);
+
+  const removeTodo = useCallback(async (todoId: number) => {
+    setDeletingTodo(todoId);
+
+    try {
+      await deleteTodo(todoId);
+
+      await loadTodos();
+    } catch {
+      handleError('Unable to delete a todo');
+    }
+
+    setDeletingTodo(-1);
```

Comment:
i don`t understand why -1?)

Reply:
what if there will be 0 as id hah))) today Sasha said that it is mostly impossible, but I didn't know tat when I was writing code)